src/blosc2/proxy.py

```diff
@@ -166,6 +185,50 @@ async def afetch(self, item=None):
         -----
         This method is only available if the :ref:`ProxySource` has an
         async `aget_chunk` method.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> import blosc2
+        >>> import asyncio
+        >>> class MyProxySource:
+        >>>       def __init__(self, data):
+        >>>                # If the source is multidimensional, it must have the attributes:
+        >>>                 self.data = data
+        >>>                 self.shape = data.shape
+        >>>                 self.chunks = [data.shape[i] for i in range(data.ndim)]
+        >>>                 self.blocks = [data.shape[i] for i in range(data.ndim)]
+        >>>                 self.dtype = data.dtype
+        >>>                 f"Data shape: {self.shape}, Chunks: {self.chunks}"
+        Data shape: (4, 5), Chunks: [4, 5]
+        >>>                 f"Blocks: {self.blocks}, Dtype: {self.dtype}"
+        Blocks: [4, 5], Dtype: int64
+        >>>       async def aget_chunk(self, nchunk):
+        >>>                 await asyncio.sleep(0.1)
+        >>>                 return self.data[nchunk]
+        >>> # Class that inherits from blosc2.Proxy
+        >>> class MyProxy(blosc2.Proxy):
+        >>>       def __init__(self, source):
+        >>>                 super().__init__(source)
+        >>>                 self._cache = source.data
+        >>>       # Asynchronous method to get the cache data
+        >>>       async def afetch(self, slice_=None):
+        >>>                 return self._cache if slice_ is None else self._cache[slice_]
+        >>> data = np.arange(20).reshape(4, 5)
+        >>> source = MyProxySource(data)
+        >>> proxy = MyProxy(source)
+        >>> async def fetch_data():
+        >>>         full_data = await proxy.afetch()
+        >>>         f"Full data cache: {full_data[:]}"
+        Full data cache:    [[ 0  1  2  3  4]
+                            [ 5  6  7  8  9]
+                            [10 11 12 13 14]
+                            [15 16 17 18 19]]
+        >>>         slice_data = await proxy.afetch(slice(0, 2))
+        >>>         f"Slice data cache: {slice_data[:]}"
```

Comment:
Again, it would much more illustrative if you invert the data fetches, and always dump the complete proxy (showing that, only when all data has been fetched, the dump is going to be identical to source).